pkg/client/cache/store.go

```diff
@@ -76,6 +81,57 @@ func (c *cache) Add(obj interface{}) error {
 	c.lock.Lock()
 	defer c.lock.Unlock()
 	c.items[key] = obj
+	c.updateIndices(obj)
```

Comment:
so if on update, i get the previous object based on its key, and then remove that object from all indices, then for updated object treat like an add, it should be fine.  will make an update this afternoon.